client/state/plans/selectors/plan.ts

```diff
@@ -1,38 +1,31 @@
 import { createSelector } from '@automattic/state-utils';
 import { get, find } from 'lodash';
+import type { PricedAPIPlan } from '@automattic/data-stores';
+import type { AppState } from 'calypso/types';
 
 import 'calypso/state/plans/init';
 
 /**
  * Return WordPress plans getting from state object
- *
- * @param {object} state - current state object
- * @returns {Array} WordPress plans
  */
-export const getPlans = ( state ) => {
+export const getPlans = ( state: AppState ): PricedAPIPlan[] => {
 	return state.plans.items;
 };
 
 /**
  * Return requesting state
- *
- * @param {object} state - current state object
- * @returns {boolean} is plans requesting?
  */
-export const isRequestingPlans = ( state ) => {
+export const isRequestingPlans = ( state: AppState ): boolean => {
 	return state.plans.requesting;
 };
 
 /**
  * Returns a plan
- *
- * @param  {object} state      global state
- * @param  {number} productId  the plan productId
- * @returns {object} the matching plan
  */
 export const getPlan = createSelector(
-	( state, productId ) => find( getPlans( state ), { product_id: productId } ),
-	( state ) => getPlans( state )
+	( state: AppState, productId: string | number ): PricedAPIPlan | undefined =>
+		getPlans( state ).find( ( plan ) => plan.product_id === productId ),
+	( state: AppState ) => getPlans( state )
```

Comment:
We received an error report where `getPlans(...)` isn't returning an array and so `.find` is throwing an error. The lodash `find` function likely played nicely with the undefined result, so we just need to make sure we handle it here as well.

Reply:
We're working on a quick fix and should have a PR shortly.

Reply:
This was my fault in the work I did in https://github.com/Automattic/wp-calypso/pull/72032

The fix is here: https://github.com/Automattic/wp-calypso/pull/72878